src/python/modules/ensembl/compara/plugins/TestDB.py

```diff
@@ -0,0 +1,249 @@
+
+    @staticmethod
+    def _get_sql_filter(filter_by: Union[str, List, None]) -> str:
+        """Returns an SQL WHERE clause including all the given conditions.
+
+        If more than one condition is given, they will be joined by the AND operator and put inside
+        parenthesis. Symbols that can be considered placeholders by sqlalchemy (like ``%``) will be doubled
+        (``%%``) to keep their initial meaning.
+
+        Args:
+            filter_by: Condition(s) to be combined in the WHERE statement.
+
+        Returns:
+            An empty string if `filter_by` is empty, the SQL WHERE clause otherwise.
+
+        """
+        sql_filter = ""
+        if filter_by:
+            if isinstance(filter_by, list):
+                filter_by = ") AND (".join(filter_by)
+            # Single percentage symbols ("%") are interpreted as placeholders, so double them up
+            filter_by = re.sub(r'(?<!%)%(?!%)', '%%', filter_by)
+            sql_filter = "WHERE ({})".format(filter_by)
+        return sql_filter
+
+    def test_num_rows(self, variation: float = 0.0, group_by: Union[str, List] = None,
+                      filter_by: Union[str, List] = None) -> None:
+        """Compares the number of rows between reference and target tables.
+
+        If `group_by` is provided, the number of rows will be compared per group, applying the same variation
+        to all of them. If `filter_by` is provided, only the rows matching all the given conditions will be
+        compared.
+
+        Note:
+            The ceiling function is applied to round the allowed variation in order to compare two integers.
+            Thus, the test may pass even if the difference between two files is greater than the exact allowed
+            variation, e.g. the test will pass if the difference is 2 and the allowed variation is 1,4.
+
+        Args:
+            variation: Allowed variation between reference and target tables.
+            group_by: Group rows by column(s), and count the number of rows per group.
+            filter_by: Filter rows by one or more conditions (joined by the AND operator).
+
+        Raise:
+            TestDBException: If `group_by` is provided and the number of groups is different; or if the number
+                of rows differ for at least one group.
+
+        """
+        # Compose the sql query from the given parameters
+        sql_filter = self._get_sql_filter(filter_by)
+        if group_by:
+            if (group_by is None) or isinstance(group_by, list):
+                group_by = ", ".join(group_by)
+            # ORDER BY to ensure that the results are always in the same order (for the same groups)
+            sql_query = "SELECT {0}, COUNT(*) as nrows FROM {1} {2} GROUP BY {0} ORDER BY {0}".format(
+                group_by, self.table, sql_filter)
+        else:
+            sql_query = "SELECT COUNT(*) as nrows FROM {} {}".format(self.table, sql_filter)
```

Comment:
I do usually prefer not to add parameters I'm not going to use, as I find it may affect the readability of the code. I'll wait to see if someone wants to add their opinion regarding this.